zeppelin-web/src/components/notebookListDataFactory/notebookList.datafactory.js

```diff
@@ -13,14 +13,28 @@
  */
 'use strict';
 
-angular.module('zeppelinWebApp').factory('notebookListDataFactory', function() {
+angular.module('zeppelinWebApp').factory('notebookListDataFactory', function($location) {
   var notes = {};
 
   notes.list = [];
 
   notes.setNotes = function(notesList) {
+  	if(notes.list.length && notes.list.length+1===notesList.length){
+		var newNoteID = getNewNoteID(notesList);
```

Comment:
@felixcheung Yes, there is a possible race condition when there multiple connections.
@MikeTYChen I think the `broadcastNote` method in NotebookServer.java isn't sending the new notebook object to the current socket. `getNewNoteID` should use the id from that response instead of picking it from the list. Thank you.